src/EditorFeatures/Core/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -301,7 +300,7 @@ internal class CodeActionEditHandlerService : ForegroundThreadAffinitizedObject,
             {
                 var navigationService = workspace.Services.GetRequiredService<IDocumentNavigationService>();
                 await navigationService.TryNavigateToPositionAsync(
-                    workspace, navigationOperation.DocumentId, navigationOperation.Position, cancellationToken).ConfigureAwait(false);
+                    this.ThreadingContext, workspace, navigationOperation.DocumentId, navigationOperation.Position, cancellationToken).ConfigureAwait(false);
```

Comment:
teh general concept here is that all the helpers which actually do perform a navigation take a ThreadingCOntext.  That way they *could*, if necessary, switch to a UI thread.  This also means taht the helpers are only available to consumer who ahve a threading context and thus live in an 'editor' world where the concepts of navigating actually make sense.

Reply:
Ah, interesting way to split that up. I think I like it!